src/main/java/org/solid/testharness/utils/SolidResourceProvider.java

```diff
@@ -135,11 +131,10 @@ public AccessDataset getAccessDataset() throws Exception {
         return getAclUrl() != null ? solidClientProvider.getAcl(aclUrl) : null;
     }
 
-    public boolean setAccessDataset(final AccessDataset accessDataset) throws Exception {
-        if (Boolean.FALSE.equals(aclLinkAvailable)) return false;
+    public void setAccessDataset(final AccessDataset accessDataset) throws Exception {
         final URI aclUrl = getAclUrl();
-        if (aclUrl == null) return false;
-        return solidClientProvider.createAcl(aclUrl, accessDataset);
+        if (aclUrl == null) throw new Exception("Failed to find ACL Link for this resource");
```

Comment:
Why not `TestHarnessException` here?

Reply:
Because that exception is a Runtime exception only used in the API layer and this is the internal CTH layer. I think there is a case for and internal custom exception but that would be for a different task.